pkg/provider/azure_loadbalancer.go

```diff
@@ -1820,31 +1821,53 @@ func (az *Cloud) reconcileFrontendIPConfigs(clusterName string, service *v1.Serv
 	lbFrontEndIPConfigNewConfigs := []network.FrontendIPConfiguration{}
 	changedTotal := false
 	var subnet network.Subnet
+
+	newConfigs := map[bool][]network.FrontendIPConfiguration{}
+	pipRG := az.getPublicIPAddressResourceGroup(service)
+	// Put all LB's FIPs into newConfigs according to IP family
+	if lb.FrontendIPConfigurations != nil {
+		for i := range *lb.FrontendIPConfigurations {
+			fip := (*lb.FrontendIPConfigurations)[i]
+			fipIsIPv6, err := az.isFIPIPv6(pipRG, &fip)
+			if err != nil {
+				return ownedFIPConfigs, toDeleteConfigs, changedTotal, err
+			}
+			newConfigs[fipIsIPv6] = append(newConfigs[fipIsIPv6], fip)
+		}
+	}
+
 	handleFrontendIPConfig := func(isIPv6 bool) error {
-		ownedFIPConfig, toDeleteConfigsSingleStack, changed, newConfigs, err := az.reconcileFrontendIPConfigsSingleStack(clusterName, service, lb, lbStatus, wantLb, isIPv6, lbFrontendIPConfigName[isIPv6], &subnet)
+		ownedFIPConfig, toDeleteConfigsSingleStack, changed, newConfigs, err := az.reconcileFrontendIPConfigsSingleStack(clusterName, service, lb, lbStatus, wantLb, isIPv6, newConfigs[isIPv6], lbFrontendIPConfigName[isIPv6], &subnet)
 		if err != nil {
 			return err
 		}
 
 		if changed {
-			lbFrontEndIPConfigNewConfigs = append(lbFrontEndIPConfigNewConfigs, newConfigs...)
 			changedTotal = true
 		}
+		lbFrontEndIPConfigNewConfigs = append(lbFrontEndIPConfigNewConfigs, newConfigs...)
 		ownedFIPConfigs = append(ownedFIPConfigs, ownedFIPConfig)
 		toDeleteConfigs = append(toDeleteConfigs, toDeleteConfigsSingleStack...)
 		return nil
 	}
+	// Even if one IP family is not enabled for the Service, we still need to put
+	// FIPs of that IP family into lbFrontEndIPConfigNewConfigs.
 	v4Enabled, v6Enabled := getIPFamiliesEnabled(service)
 	if v4Enabled {
 		if err := handleFrontendIPConfig(false); err != nil {
 			return ownedFIPConfigs, toDeleteConfigs, changedTotal, err
 		}
+	} else {
+		lbFrontEndIPConfigNewConfigs = append(lbFrontEndIPConfigNewConfigs, newConfigs[false]...)
```

Comment:
What does this "else" mean?

Reply:
As I commented above, else means "we still need to put FIPs of that IP family into lbFrontEndIPConfigNewConfigs".